### ISSUE TITLE
Proxy per route based on Rule

### ISSUE THREAD

**Issue by sateeshd7** [mockoon/mockoon#363]
**Is your feature request related to a problem? Please describe.**
We use Mockoon especially to handle the Negative scenarios for the external API Integrations and with the help of mockoon would like handle both positive and negative scenarios based on Rules to forward the request to Proxy or not.

**Describe the solution you'd like**
The same way Status, Headers and Responses are provided per Response based on Rules, we can have proxy config as well.

**Describe alternatives you've considered**
The same kind of feature available in Mock-server and it is very useful.

**Comment by leolangen** [comment]
I could also use this feature, as we cannot stub successful responses easily for our testing purposes. Having the ability to either stub out a negative scenario or forward the request to the proxied server for the positive scenario based on rules would be ideal.

**Comment by aevalo** [comment]
+1 from me also. Ability to mock fix certain IDs to mocked responses and leave correct data to real server would benefit me greatly.

**Comment by everettcomstock** [comment]
Is there a roadmap or anyway to glean insight on what's up next for Mockoon? I'm trying to get my organization to adopt Mockoon but this one feature has become a sticking point.

**Comment by kajweb** [comment]
Yes, I have a scene now.
All requests are made to /api. 
Then the actual return value will be determined by the parameters of the request


```bash 
curl 'http://mock.ct.com/apiv3'  --data-raw '{"Action":"showA"}'

# You are requesting showA
```

```bash 
curl 'http://mock.ct.com/apiv3'  --data-raw '{"Action":"showB"}'

# You are requesting showB
```


```bash 
curl 'http://mock.ct.com/apiv3'  --data-raw '{"Action":"showC"}'

# You are requesting showC
```



If not within my rules, make a real request

**Comment by dethell-jh** [comment]
This is a huge feature for us. Now that I see the PR is pending I'm going to push for mockoon to replace our current mock/proxy solution.

**Comment by 255kb** [comment]
📦 This is now available with [release v4.1.0](https://mockoon.com/releases/4.1.0).
